h/static/scripts/annotation-ui.js

```diff
@@ -52,7 +52,10 @@ function initialState(settings) {
 
     filterQuery: null,
 
-    sortMode: 'Location',
+    // Key by which annotations are currently sorted.
+    sortKey: 'Location',
+    // Keys by which annotations can be sorted.
+    sortKeysAvailable: ['Newest', 'Oldest', 'Location'],
```

Comment:
I've turned this from an object into a couple of top-level keys. Flat structures are much easier to wrap your head around than nested ones. Worth keeping in mind, I think.